src/MPI.jl

```diff
@@ -4,21 +4,31 @@ module MPI
 
 using Compat
 
-const depfile = joinpath(dirname(@__FILE__), "..", "deps", "src", "compile-time.jl")
-isfile(depfile) || error("MPI not properly installed. Please run Pkg.build(\"MPI\")")
+@windows_only begin
+	const depfile = "win_mpiconstants.jl"
+end
+
+@unix_only begin
+	const depfile = joinpath(dirname(@__FILE__), "..", "deps", "src", "compile-time.jl")
+	isfile(depfile) || error("MPI not properly installed. Please run Pkg.build(\"MPI\")")
+end
+
 include(depfile)
+
 include("mpi-base.jl")
 include("cman.jl")
 
 function __init__()
-    # need to open libmpi with RTLD_GLOBAL flag for Linux, before any ccall
-    # cannot use RTLD_DEEPBIND; this leads to segfaults at least on Ubuntu 15.10
-    @eval const libmpi_handle =
-        Libdl.dlopen(libmpi, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
-
-    # look up all symbols ahead of time
-    for (jname, fname) in _mpi_functions
-        eval(:(const $jname = Libdl.dlsym(libmpi_handle, $fname)))
+    @unix_only begin
+        # need to open libmpi with RTLD_GLOBAL flag for Linux, before any ccall
+        # cannot use RTLD_DEEPBIND; this leads to segfaults at least on Ubuntu 15.10
+        @eval const libmpi_handle =
+            Libdl.dlopen(libmpi, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
+
+        # look up all symbols ahead of time
+        for (jname, fname) in _mpi_functions
+            eval(:(const $jname = Libdl.dlsym(libmpi_handle, $fname)))
```

Comment:
This line looks up the symbols for the MPI functions when the MPI package is loaded. Symbol lookup requires string operations and is thus comparatively expensive. It seems that, for Windows, you instead use a (symbol, library name) pair, which will perform the lookup at run time, every time an MPI function is called.

(I have heard that Julia is supposed to cache the lookups, and perform them only once -- but when I measured this, this wasn't the case.)

You might want to check this, e.g. by profiling calling `MPI_Comm_rank` in a tight loop, to see how expensive one call is.

This is not a show-stopper, but if you find there is a problem, you could add a comment to the code and open a ticket for this, as to-do item for the future.